### PR TITLE
Fix coverage on clang builds

### PR DESCRIPTION
**Pull request by fjricci**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -58,7 +71,7 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang"
   endif ()
 
   if (COVERAGE)
-    set(COMMON_FLAGS "${COMMON_FLAGS} -g -fprofile-arcs -ftest-coverage")
+    set(COMMON_FLAGS "${CMAKE_CXX_FLAGS} -g -fprofile-arcs -ftest-coverage")
```
Comment (sas) [inline CMakeLists.txt]: Why are you doing this?

File: .travis.yml [hunk 2]
```diff
@@ -48,7 +48,7 @@ matrix:
     - os: linux
       env: TARGET="Linux-X86"      LLDB_TESTS="all"  CLANG="0" COVERAGE="1" PLATFORM="1"
     - os: linux
-      env: TARGET="Linux-X86"      LLDB_TESTS="all"  CLANG="0" RELEASE="1"
+      env: TARGET="Linux-X86"      LLDB_TESTS="all"  CLANG="0" COVERAGE="1"  RELEASE="1"
     - os: linux
       env: TARGET="Linux-X86"      LLDB_TESTS="all"  CLANG="1"
```
Comment (sas) [inline .travis.yml]: Why don't we enable coverage here?

File: .travis.yml [hunk 3]
```diff
@@ -57,20 +57,20 @@ matrix:
     - os: linux
       env: TARGET="Linux-X86_64"   LLDB_TESTS="all"  CLANG="0" COVERAGE="1"
```
Comment (sas) [inline .travis.yml]: And here.

File: .travis.yml (hunk 2 above)
Comment (fjricci) [inline .travis.yml]: For some reason, libprofile for i386 clang doesn't want to install. I can do it locally, but travis can't find the package.

**Comment by fjricci** [comment]
We can't enable coverage always, because I can't get the libprofiler install on x86 to work on Travis.

**Pull request merged**